plugins/node/opentelemetry-instrumentation-aws-lambda/test/integrations/lambda-handler.test.ts

```diff
@@ -1079,4 +1080,48 @@ describe('lambda handler', () => {
       assert.strictEqual(span.parentSpanId, undefined);
     });
   });
+
+  describe('url parsing', () => {
+    it('pulls url from api gateway rest events', async () => {
+      initializeHandler('lambda-test/sync.handler');
+      const event = {
```

Comment:
Did a quick check via logging the `event` from `REST APIGW ---> Lambda` scenario (enabled `Lambda proxy integration`)

The capitalization for some `event` fields from REST APIGW are a bit different than from HTTP APIGW. ([This doc also verifies some of the capitalization diffs for REST APIGW events
](https://docs.aws.amazon.com/lambda/latest/dg/services-apigateway.html#apigateway-example-event))

- Sampled REST APIGW event:
```
{
  ...
  path: '/',
  ...
  headers: {
    ...
    Host: '<REST_API>.execute-api.us-west-2.amazonaws.com',
    ...
    'X-Forwarded-For': '15.248.7.4',
    'X-Forwarded-Port': '443',
    'X-Forwarded-Proto': 'https'
  },
  ...
  queryStringParameters: { ghd: '356', jyi: '4568' },
  ...
}
```

- Sampled HTTP APIGW event
```
{
  ...
  rawPath: '/lambda1/route2',
  ...
  headers: {
    ...
    host: '<HTTP_API>.execute-api.us-west-2.amazonaws.com',
    ...
    'x-forwarded-for': '15.248.7.4',
    'x-forwarded-port': '443',
    'x-forwarded-proto': 'https'
  },
  queryStringParameters: { abc: '123', def: '456' },
  ...
}
```

Reply:
To support REST APIGW event, can you update the URL extraction to also account for `Host`, `X-Forwarded-Port`, and `X-Forwarded-Proto`, and then update the REST event test?

Reply:
I'll see if I can do that without making the code too crazy 😄